File: app/src/main/java/com/iug/coursehub/feature/user/my_courses_screen/lesson_content_screen/LessonContentFragment.java
package com.iug.coursehub.feature.user.my_courses_screen.lesson_content_screen;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.navigation.fragment.NavHostFragment;

import com.iug.coursehub.R;
import com.iug.coursehub.data.local.db.entity.Lesson;
import com.iug.coursehub.databinding.FragmentLessonContentBinding;
import com.iug.coursehub.feature.user.my_courses_screen.lesson_list_screen.LessonListFragment;

public class LessonContentFragment extends Fragment {

    private final LessonContentViewModel viewModel = new LessonContentViewModel();

    private Integer courseId;
    private Integer lessonId;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            courseId = getArguments().getInt(ARG_COURSE_ID, 0);
            lessonId = getArguments().getInt(ARG_LESSON_ID, 0);
        }
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        viewModel.getLesson(lessonId);

        FragmentLessonContentBinding binding = FragmentLessonContentBinding.inflate(inflater, container, false);

        viewModel.lessonContent.observe(getViewLifecycleOwner(), lesson -> {
            if (lesson == null) return;

            fillFields(binding, lesson);
        });

        Bundle bundle = new Bundle();
        bundle.putInt(LessonListFragment.ARG_COURSE_ID, courseId);
        binding.btnNavigateUp.setOnClickListener(v -> NavHostFragment.findNavController(LessonContentFragment.this)
                .navigate(
                        R.id.action_lessonContentFragment_to_lessonListFragment,
                        bundle
                ));

        return binding.getRoot();
    }

    private void fillFields(FragmentLessonContentBinding binding, Lesson lesson) {
        binding.tvLessonTitle.setText(lesson.getTitle());
        binding.tvLessonDescription.setText(lesson.getDescription());
        binding.btnOpenYTVideo.setOnClickListener(v -> {
            String link = lesson.getYtLink();
            Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(link));
            startActivity(intent);
        });
    }

    public static final String ARG_COURSE_ID = "course_id";
    public static final String ARG_LESSON_ID = "lesson_id";
}

File: app/src/main/java/com/iug/coursehub/feature/user/my_courses_screen/lesson_content_screen/LessonContentFragment.kt
package com.iug.coursehub.feature.user.my_courses_screen.lesson_content_screen

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.navigation.fragment.findNavController
import com.iug.coursehub.R
import com.iug.coursehub.data.local.db.entity.Lesson
import com.iug.coursehub.databinding.FragmentLessonContentBinding
import com.iug.coursehub.feature.user.my_courses_screen.lesson_list_screen.LessonListFragment

class LessonContentFragment : Fragment() {

    private val viewModel by viewModels<LessonContentViewModel>()

    private var courseId: Int? = null
    private var lessonId: Int? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        arguments?.let {
            courseId = it.getInt(ARG_COURSE_ID, 0)
            lessonId = it.getInt(ARG_LESSON_ID, 0)
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        viewModel.getLesson(lessonId!!)

        val binding = FragmentLessonContentBinding.inflate(inflater, container, false)

        binding.apply {
            viewModel.lessonContent.observe(viewLifecycleOwner) { lesson ->
                if (lesson == null)
                    return@observe

                this.fillFields(lesson)
            }

            btnNavigateUp.setOnClickListener {
                findNavController().navigate(
                    R.id.action_lessonContentFragment_to_lessonListFragment,
                    Bundle().apply {
                        putInt(LessonListFragment.ARG_COURSE_ID, courseId!!)
                    }
                )
            }
        }

        return binding.root
    }

    private fun FragmentLessonContentBinding.fillFields(lesson: Lesson) {
        tvLessonTitle.text = lesson.title
        tvLessonDescription.text = lesson.description
        btnOpenYTVideo.setOnClickListener {
            val link = lesson.ytLink
            val intent = Intent(Intent.ACTION_VIEW, Uri.parse(link))
            startActivity(intent)
        }
    }

    companion object {
        const val ARG_COURSE_ID = "course_id"
        const val ARG_LESSON_ID = "lesson_id"
    }
}